BigImageViewer/src/main/java/com/github/piasy/biv/view/BigImageView.java

```diff
@@ -503,31 +526,65 @@ public void onAnimationRepeat(Animation animation) {
 
     @UiThread
     private void doShowImage(final int imageType, final File image) {
-        if (mMainView != null) {
-            removeView(mMainView);
-        }
+        doShowImage(imageType, image, false);
+    }
 
-        mMainView = mViewFactory.createMainView(getContext(), imageType, image, mInitScaleType);
-        if (mMainView == null) {
-            onFail(new RuntimeException("Image type not supported: "
-                                        + ImageInfoExtractor.typeName(imageType)));
-            return;
-        }
+    @UiThread
+    private void doShowImage(final int imageType, final File image, final boolean useThumbnailView) {
 
-        addView(mMainView, ViewGroup.LayoutParams.MATCH_PARENT,
-                ViewGroup.LayoutParams.MATCH_PARENT);
-        mMainView.setOnClickListener(mOnClickListener);
-        mMainView.setOnLongClickListener(mOnLongClickListener);
+        if (useThumbnailView) {
 
-        if (mMainView instanceof SubsamplingScaleImageView) {
-            mSSIV = (SubsamplingScaleImageView) mMainView;
+            if(mThumbnailView != null) {
+                removeView(mThumbnailView);
+            }
+
+            mThumbnailView = new ImageView(getContext());
```

Comment:
I have a question here: is it mandatory that the shared view is an `ImageView` to make shared animation work?

Reply:
I don't think it's mandatory, but I'd have to test to be sure...
Do you want to use SSIV?

Reply:
If it't not mandatory, I think it's better to use `mViewFactory.createThumbnailView` to create the thumbnail view.

Reply:
Wait a sec, that function returns null, so do you want me to make it return a standard ImageView or what? I'm confused...

Reply:
The base class `ImageViewFactory` returns null, but the actual object would be instance of `FrescoImageViewFactory` or `GlideImageViewFactory`, which won't return null.

But maybe we could return a `ImageView` in `ImageViewFactory#createThumbnailView`.

Reply:
Ok done, I had to returned an ImageView, otherwise this scheme would only be usable with FrescoImageViewFactory or GlideImageViewFactory, tested and working.